nautobot_secrets_providers/providers/one_password.py

```diff
@@ -0,0 +1,111 @@
+        section (str, optional): 1Password Item Section for the secret. Defaults to None.
+
+    Returns:
+        (str): Value from the secret.
+    """
+    client = await Client.authenticate(
+        auth=token, integration_name="nautobot-secrets-providers", integration_version=__version__
+    )
+    reference = f"op://{vault}/{item}/{f'{section}/' if section else ''}{field}"
+    return await client.secrets.resolve(reference)
+
+
+def vault_choices():
+    """Generate Choices for vault form field.
+
+    Build a form option for each key in vaults.
+    """
+    plugin_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"]
+    return [(key, key) for key in plugin_settings["one_password"]["vaults"].keys()]
+
+
+class OnePasswordSecretsProvider(SecretsProvider):
+    """A secrets provider for 1Password."""
+
+    slug = "one-password"
+    name = "1Password Vault"
+    is_available = Client is not None
+
+    # TBD: Remove after pylint-nautobot bump
+    # pylint: disable-next=nb-incorrect-base-class
+    class ParametersForm(BootstrapMixin, forms.Form):
+        """Required parameters for HashiCorp Vault."""
+
+        vault = forms.ChoiceField(
+            required=True,
+            choices=vault_choices,
+            help_text="1Password Vault to retrieve the secret from.",
+        )
+        item = forms.CharField(
+            required=True,
+            help_text="The item in 1Password.",
+        )
+        section = forms.CharField(
+            required=False,
+            help_text="The section where the field is a part of.",
+        )
+        field = forms.CharField(
+            required=True,
+            help_text="The field where the secret is located. Defaults to 'password'.",
+            initial="password",
+        )
+
+    @classmethod
+    def get_token(cls, secret, vault):
+        """Get the token for a vault."""
+        plugin_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"]
+        if "token" in plugin_settings["one_password"]["vaults"][vault]:
+            return plugin_settings["one_password"]["vaults"][vault]["token"]
+        try:
+            return plugin_settings["one_password"]["token"]
+        except KeyError as exc:
+            raise exceptions.SecretProviderError(secret, cls, "1Password token is not configured!") from exc
+
+    @classmethod
+    def get_value_for_secret(cls, secret, obj=None, **kwargs):  # pylint: disable=too-many-locals
+        """Get the value for a secret from 1Password."""
+        # This is only required for 1Password therefore not defined in
+        # `required_settings` for the app config.
+        plugin_settings = settings.PLUGINS_CONFIG["nautobot_secrets_providers"]
+        if "one_password" not in plugin_settings:
+            raise exceptions.SecretProviderError(secret, cls, "1Password is not configured!")
+
+        parameters = secret.rendered_parameters(obj=obj)
+        vault = parameters["vault"]
+
+        return asyncio.run(
```

Comment:
Instead of instantiating our own event loop should we be using [Django's async/sync wrappers](https://docs.djangoproject.com/en/5.1/topics/async/#async-to-sync)? It seems like those exist to prevent us from running into weird multiprocessing issues?